exporter/exporterhelper/common.go

```diff
@@ -277,11 +288,7 @@ func newBaseExporter(set exporter.Settings, signal component.DataType, osf obsre
 
 	be.connectSenders()
 
-	if bs, ok := be.batchSender.(*batchSender); ok {
-		// If queue sender is enabled assign to the batch sender the same number of workers.
-		if qs, ok := be.queueSender.(*queueSender); ok {
-			bs.concurrencyLimit = int64(qs.numConsumers)
```

Comment:
The biggest problem with this approach is that we ignore the number of queue workers. It means that, typically, every batch would have to be stuck for the flush_timeout period before it can be sent further. That's why we had the concurrency limit built tied to it, so every time all the workers are stuck in one batch, we release them right away. And that behavior is gone with this PR.

Reply:
So, it doesn't resolve the problem stated in https://github.com/open-telemetry/opentelemetry-collector/issues/10368. Just adding an independent concurrency limit doesn't help for that

Reply:
@dmitryax Thanks for the feedback I've spent a bit of time thinking about options here so that the queue_sender consumers don't block when there are more items we can batch. I’m wondering if we can swap the order of the batch sender and the queue sender so that we are actually queueing batches instead of queuing the initial request that comes in? Or otherwise it seems that the queue sender could do some prebatching in the queue_sender (i.e. keep popping items from the queue until we have roughly a batch size to send to the batch_sender).

Reply:
Another possibility I explored was unblocking consumers by calling [consumeFunc](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/internal/queue/bounded_memory_queue.go#L52) in a goroutine so that the batch_sender isn't blocked for the flush timeout and more requests can be consumed. However this might not work for the persistent queue which needs to know when `consumeFunc` returns so it can mark items for deletion from the persistent queue.

Reply:
Any thoughts on these options?

Reply:
A few problems with putting the queue before the batcher:
- The queue is what adds asynchronous behavior to exporters. If it's not enabled, the batcher is expected to concurrently block the incoming requests.  
- Persistent queue provides durability. A crashed collector is expected to not lose any data. If we put another memory queue in front of batching, it breaks that promise.

Reply:
> Another possibility I explored was unblocking consumers by calling [consumeFunc](https://github.com/open-telemetry/opentelemetry-collector/blob/main/exporter/internal/queue/bounded_memory_queue.go#L52) in a goroutine so that the batch_sender isn't blocked for the flush timeout and more requests can be consumed. However this might not work for the persistent queue which needs to know when consumeFunc returns so it can mark items for deletion from the persistent queue.

I've been coming to this conclusion. I'm imagining it would be something like the following:
 - batcher reads, but does not remove, records from the queue until it has a batch of min size
 - batcher sends the batch to the next consumer in the chain -- this could be the concurrency sender
 - on success, batcher removes the records from the queue

Perhaps there could be an option on the batcher to remove records from the queue optimistically, for users that prefer at most once vs. at least once delivery (default being at least once, for durability).

This is not trivial of course - it would mean a substantial change to the queue API.